testing-common/src/test/groovy/context/FieldBackedProviderTest.groovy

```diff
@@ -177,8 +180,12 @@ class FieldBackedProviderTest extends AgentTestRunner {
 
   def "context classes are redefine safe"() {
     when:
-    ByteBuddyAgent.getInstrumentation().redefineClasses(new ClassDefinition(KeyClass, ClasspathUtils.convertToByteArray(KeyClass)))
-    ByteBuddyAgent.getInstrumentation().redefineClasses(new ClassDefinition(UntransformableKeyClass, ClasspathUtils.convertToByteArray(UntransformableKeyClass)))
+    try {
```

Comment:
@trask This started throwing an exception even though the assertions pass, any idea if this is a valid fix?

Reply:
It seems strange, that test named "context classes are redefine safe" is expected to throw an exception.

Reply:
i don't think it's a valid fix 😄

i tried it out, looks like jacoco instrumentation is getting applied to `KeyClass` and `UntransformableKeyClass` in your branch, which is causing the redefinition to fail, e.g.:

```
  def "context classes are redefine safe"() {
    when:
    for (Field field : KeyClass.getDeclaredFields()) {
      println field.getName()
    }
```

shows `$jacocoData` in your branch, but no jacoco field in master.

it looks like those classes are supposed to be excluded from jacoco:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/testing-common/testing-common.gradle#L13-L14

so maybe your changes broke that exclusion somehow?

Reply:
Thanks for the debugging! Yeah that must be it I see what needs to be fixed. For reference though, does this mean that this behavior is only guaranteed when there isn't another agent running? I wonder how many of our features have that restriction.

Reply:
Also I don't know why `UntransformableKeyClass` is not transformable that would be helpful if you know :)